x/wasm/internal/keeper/handler_plugin.go

```diff
@@ -170,6 +180,20 @@ func EncodeStakingMsg(sender sdk.AccAddress, msg *wasmvmtypes.StakingMsg) ([]sdk
 	}
 }
 
+func EncodeStargateMsg(unpacker codectypes.AnyUnpacker) StargateEncoder {
+	return func(sender sdk.AccAddress, msg *wasmvmtypes.StargateMsg) ([]sdk.Msg, error) {
+		any := codectypes.Any{
+			TypeUrl: msg.TypeURL,
+			Value:   msg.Value,
+		}
+		var sdkMsg sdk.Msg
+		if err := unpacker.UnpackAny(&any, &sdkMsg); err != nil {
+			return nil, sdkerrors.Wrap(types.ErrInvalidMsg, fmt.Sprintf("Cannot unpack proto message with type URL: %s", msg.TypeURL))
+		}
+		return []sdk.Msg{sdkMsg}, nil
```

Comment:
The `sdkMsg` can contain `Any` types as well that need to be unpacked. This is not done recursively (https://github.com/cosmos/cosmos-sdk/blob/v0.41.4/codec/types/compat.go#L197).
You need another call to `codectypes.UnpackInterfaces(sdkMsg, unpacker)`

Reply:
I did this here: https://github.com/CosmWasm/wasmd/pull/435/commits/c33b3694acef890e5754e2e0e75ac0bf9a701a14

Can you check that was correct?

Reply:
ja, looks good!
btw. you can use `sdkerrors.Wrapf` to save you some `fmt.Sprintf`